core/editor/src/tools/mod.rs

```diff
@@ -1,14 +1,30 @@
-use crate::events::{ModKeys, MouseState, TracePoint};
-use crate::{events::Trace, Color};
+mod crop;
+mod ellipse;
+mod line;
+mod navigate;
+mod path;
+mod pen;
+mod rectangle;
+mod sample;
+mod select;
+mod shape;
+
+use crate::events::{Event, ModKeys, MouseState, Trace, TracePoint};
+use crate::Color;
 use std::collections::HashMap;
 
+pub trait Tool {
+	fn handle_input(&mut self, event: Event);
```

Comment:
I think the function should also return an `Operation`
The Operation would need to be defined in the `document-core` and would just be an enum with values like AddCircle(center, radius, color, …)
But we could also put that into a seperate PR if it is too much work

Reply:
I will do that, I'm wondering if it should return `Option<Operation>` instead though.

Reply:
Either that or add a none operation but I think an option would be cleaner